### DISCUSSION TITLE
built project but npm start fails

### DISCUSSION THREAD

**Discussion by GiulioCesare** [RussCoder/djvujs#81] [Q&A] [answered]
Hello RussCoder, my name is Giulio and I was looking into this project and tried to build and test it.
I followed the steps to build the project and everything seems fine.
But where I run "npm start" from the viewer directory I get this error in the browser:

Error: There is no DjVu object! You have to include the DjVu.js library first!
./src/DjVu.js
http://localhost:3000/static/js/main.chunk.js:18:9

  15 |  * Also it encapsulates the logic of getting the global DjVu object.
  16 |  */
  17 | if (typeof DjVu !== 'object') {
> 18 |   throw new Error("There is no DjVu object! You have to include the DjVu.js library first!");
     |         ^  19 | }
  20 | 
  21 | const djvu = DjVu; // eslint-disable-line

Plus other errors

Any idea why? 

Many thanks for any help

**Comment by r-pankevicius** [comment]
I think I saw it. You need to include DjVu.js library in your main html page.

If you go to view-source:https://djvu.js.org/ then you'll see this sequence of includes:
```js
<script src="[/assets/dist/djvu.js](view-source:https://djvu.js.org/assets/dist/djvu.js)"></script
<script src="[/assets/dist/djvu_viewer.js](view-source:https://djvu.js.org/assets/dist/djvu_viewer.js)">
```
djvu.js is a library, djvu_viewer.js is a viewer and .. read again: "You have to include the DjVu.js library first!"

**Reply by GiulioCesare** [reply]
Thank you r-pankevicius. The problem was that I might have missed the  "npm run install" step.
I did as suggested by RussCoder and now it works.

**Comment by RussCoder** [comment, marked as answer]
Hello, Giulio!

That's because you started the viewer without building the library at least once. 

In the **root directory of the project** run:

```
npm run install
npm run build
```

It will build everything. Then you can start the viewer and work only with it. 
Alternatively you can build only the library, in the `library` folder run `npm install` and `npm run build` - the library bundle will be copied to the viewer's directory. If you want to change the library then run `npm start` there - the library will be recreated on each change. 

The library and the viewer have separate configs, so you should either build the library once or run `npm start` in two consoles - one for the library and one for the viewer.

**Reply by GiulioCesare** [reply]
Thank you RussCoder for the prompt reply.
I might have missed one of the above steps.
I did as suggested and it now works.
Best regards
